meson.build

```diff
@@ -669,8 +669,13 @@ conf.set_quoted('DEFAULT_NET_NAMING_SCHEME', default_net_naming_scheme)
 
 time_epoch = get_option('time-epoch')
 if time_epoch == -1
-        NEWS = files('NEWS')
-        time_epoch = run_command(stat, '-c', '%Y', NEWS).stdout().to_int()
+        source_date_epoch = run_command('sh', ['-c', 'echo "$SOURCE_DATE_EPOCH"']).stdout().strip()
```

Comment:
urks...

hmm, this is a bit ugly... maybe we should turn both into a single heler script in tools/.

I figure @keszybz knows this stuff better than I. What's your take on this?

(Apparently meson doesn't have a getenv call natively)

Reply:
Yeah, meson does not read env by design. I'm not a meson expert, so I don't know if this is "mesonic".